### PR TITLE
Mandatory Login BE and Ruby tests

### PR DESCRIPTION
**Pull request by guillegiu**

### CODE REVIEW

File: server/controllers/ticket/create.php [hunk 1]
```diff
@@ -114,22 +122,43 @@ public function handler() {
...
+    }
 
-        if(!Controller::isUserSystemEnabled() && !Controller::isStaffLogged()) {
-            $session = Session::getInstance();
-            $session->createTicketSession($this->ticketNumber);
-        }
-        
-        Log::createLog('CREATE_TICKET', $this->ticketNumber);
+    private function isEmailInvalid(){
+        return  (Session::getInstance()->sessionExists() && User::getUser(Session::getInstance()->getUserId(),'id')  &&  $this->email && !(User::getUser(Session::getInstance()->getUserId(),'id')->email == $this->email));
```
Comment (ivandiazwm) [inline server/controllers/ticket/create.php:135]: this is too long please put `Session::getInstance()` in a variable and `Session::getInstance()->getUserId()` too

File: server/controllers/system/email-polling.php [hunk 2]
```diff
@@ -20,6 +20,8 @@ public function validations() {
     }
 
     public function handler() {
+        throw new RequestException(ERRORS::NO_PERMISSION);
```
[the remark this answers is not in the document]
Comment (guillegiu) [reply server/controllers/system/email-polling.php:23]: yep , this is intencional.You should not able to access to email-polling.Would you rather I delete the path?

File: server/controllers/ticket/check.php [hunk 3]
```diff
@@ -54,20 +54,28 @@ public function validations() {
     }
 
     public function handler() {
-        if (Controller::isUserSystemEnabled() || Controller::isStaffLogged()) {
+        if(Session::getInstance()->sessionExists())  throw new Exception(ERRORS::SESSION_EXISTS);
```
[the remark this answers is not in the document]
Comment (guillegiu) [reply server/controllers/ticket/check.php:57]: you have to logout to check other ticket

File: server/controllers/ticket/get.php [hunk 4]
```diff
@@ -32,52 +32,29 @@ class TicketGetController extends Controller {
...
-                    'ticketNumber' => [
-                        'validation' => DataValidator::equals($session->getTicketNumber()),
-                        'error' => ERRORS::INVALID_TICKET
-                    ],
-                    'csrf_token' => [
-                        'validation' => DataValidator::equals($session->getToken()),
-                        'error' => ERRORS::INVALID_TOKEN
-                    ]
+        return [
+            'permission' => 'any',
```
[the remark this answers is not in the document]
Comment (guillegiu) [reply server/controllers/ticket/get.php:36]: you are right, I'll  change it

File: server/controllers/system/disable-mandatory-login.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    public function handler() {
+        $password = Controller::request('password');
+       
+        if(!Setting::getSetting('registration')->getValue()) { 
+            throw new Exception(ERRORS::REGISTRATION_IS_DESACTIVATED);
+        }
+        
+        if(!Hashing::verifyPassword($password, Controller::getLoggedUser()->password)) {
+            throw new RequestException(ERRORS::INVALID_PASSWORD);
+            return;
```
[the remark this answers is not in the document]
Comment (guillegiu) [reply server/controllers/system/disable-mandatory-login.php:45]: Ok,  I just keeped the structure of others paths.I'll delete if you want

File: server/controllers/system/download.php [hunk 6]
```diff
@@ -57,7 +57,7 @@ public function handler() {
                     }
                     break;
                 case FileManager::PERMISSION_ARTICLE:
-                    if(Controller::isUserSystemEnabled() && !$session->sessionExists()) {
+                    if(!$session->sessionExists()) {
```
[the remark this answers is not in the document]
Comment (guillegiu) [reply server/controllers/system/download.php:60]: Why would it be necessary to verify the mandatory login? The IF checks if the system user is enabled, that will always be true. If I check the mandatory-login, the IF "logic" would be different

File: server/controllers/user/signup.php [hunk 7]
```diff
@@ -129,15 +127,19 @@ public function storeRequestData() {
...
+        $userInstance = ($user->isNull() ? new User() : $user );
+        $UserTickets = ($user->isNull() ? 0 : $user->tickets);
 
         $userInstance->setProperties([
             'name' => $this->userName,
             'signupDate' => Date::getCurrentDate(),
-            'tickets' => 0,
+            'tickets' => $UserTickets,
             'email' => $this->userEmail,
             'password' => Hashing::hashPassword($this->userPassword),
```
[the remark this answers is not in the document]
Comment (guillegiu) [reply server/controllers/user/signup.php:140]: Ticket/create sends a random token as "password" when makes a call to user/signup.line 140 works perfectly

File: server/controllers/ticket/edit-comment.php [hunk 8]
```diff
@@ -31,39 +31,19 @@ class EditCommentController extends Controller {
...
-                        'error' => ERRORS::INVALID_TOKEN
-                    ]
-                ]
-            ];
-        }
+            ]
+        ];
     }
 
     public function handler() {
```
[the remark this answers is not in the document]
Comment (guillegiu) [reply server/controllers/ticket/edit-comment.php:49]: canManageTicket  is in line 66

**Comment by guillegiu** [comment]
> What is the intention of `Controller::isUserLogged()`? It currently returns true even if you logged with a ticket.
> 
> Also I didn't check the tests

`Controller::isUserLogged()` Is used  in the user side when  you try to create a ticket and when you try to check one, if you are not logged when you try to create a ticket  it should create a new user, and check should not allow you to check another ticket if you are already with a session, already logged.

File: server/controllers/ticket/edit-comment.php (hunk 8 above)
[the remark this answers is not in the document]
Comment (guillegiu) [reply server/controllers/ticket/edit-comment.php:49]: the 'canManageTicket' is actually at line 66

File: server/controllers/user/invite.php [hunk 9]
```diff
@@ -64,9 +64,6 @@ public function validations() {
     }
 
     public function handler() {
```
[the remark this answers is not in the document]
Comment (guillegiu) [reply server/controllers/user/invite.php:66]: the path works perfectly

[the remark this answers is not in the document]
Comment (guillegiu) [reply server/controllers/user/invite.php:66]: I'll create Ruby tests to be sure

File: client/src/app/main/main-layout-header.js [hunk 10]
```diff
@@ -60,11 +95,21 @@ class MainLayoutHeader extends React.Component {
     changeLanguage(event) {
         this.props.dispatch(ConfigActions.changeLanguage(event.target.value));
     }
+
+    changeShowLoginForm(showLoginForm = true) {
+        const {
+            dispatch,
+            mandatoryLogin
+        } = this.props;
+        console.log("click: ", mandatoryLogin.showLoginForm);
```
Comment (ivandiazwm) [inline client/src/app/main/main-layout-header.js:104]: delete

File: client/src/core-components/widget-transition.scss [hunk 11]
```diff
@@ -6,10 +6,10 @@
...
         -webkit-backface-visibility: hidden;
         -webkit-transform: translate3d(0,0,0);
-        display: inline-block;
+        //display: inline-block;
         visibility: visible;
         backface-visibility: hidden;
-        position: absolute;
-        left: 0;
+        //position: absolute;
+        //left: 0;
```
Comment (ivandiazwm) [inline client/src/core-components/widget-transition.scss:13]: delete

File: client/src/app/main/main-check-ticket-page.js [hunk 12]
```diff
@@ -52,7 +52,7 @@ class MainCheckTicketPage extends React.Component {
     getClass() {
         let classes = {
             'main-check-ticket-page': true,
-            'col-md-10 col-md-offset-1': (!this.props.config['user-system-enabled'])
+            'col-md-10 col-md-offset-1': false
```
Comment (ivandiazwm) [inline client/src/app/main/main-check-ticket-page.js:55]: delete line

**Pull request merged**